src/main/scala/edu/ie3/osmogrid/io/input/AssetSource.scala

```diff
@@ -45,33 +46,32 @@ object AssetSource {
   }
 
   final case class AssetFileSource(
-      assetInformation: AssetInformation
+      executionContextExecutor: ExecutionContextExecutor,
+      csvSep: String,
+      directoryPath: String,
+      namingStrategy: FileNamingStrategy
   ) extends AssetSource {
 
-    override def read(): AssetInformation = {
-      assetInformation
-    }
-  }
-  object AssetFileSource {
-
-    def apply(
-        csvSep: String,
-        folderPath: String,
-        namingStrategy: FileNamingStrategy
-    ): AssetFileSource = {
-      val typeSource: CsvTypeSource =
-        new CsvTypeSource(csvSep, folderPath, namingStrategy)
-      val transformerTypes = typeSource.getTransformer2WTypes.asScala.toSeq
-      val lineTypes = typeSource.getLineTypes.asScala.toSeq
-      (transformerTypes, lineTypes) match {
-        case (transformerTypes, _) if transformerTypes.isEmpty =>
-          throw InputDataException(
-            s"There are no transformer types at: $folderPath"
-          )
-        case (_, lineTypes) if lineTypes.isEmpty =>
-          throw InputDataException(s"There are no line types at: $folderPath")
-        case (_, _) =>
-          AssetFileSource(AssetInformation(lineTypes, transformerTypes))
+    override def read(): Future[AssetInformation] = {
+      implicit val implicitEc: ExecutionContextExecutor =
+        executionContextExecutor
+      Future {
+        val typeSource: CsvTypeSource =
+          new CsvTypeSource(csvSep, directoryPath, namingStrategy)
+        val transformerTypes = typeSource.getTransformer2WTypes.asScala.toSeq
+        val lineTypes = typeSource.getLineTypes.asScala.toSeq
+        (transformerTypes, lineTypes) match {
+          case (transformerTypes, _) if transformerTypes.isEmpty =>
```

Comment:
This could be simplified like below (Nil is the empty list), this is of course not mandatory though.
```suggestion
          case (Nil, _) =>
```